src/contracts/adapters/linea/LineaAdapter.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {IMessageService} from './interfaces/IMessageService.sol';
+import {BaseAdapter} from '../BaseAdapter.sol';
+import {ChainIds} from 'solidity-utils/contracts/utils/ChainHelpers.sol';
+import {Errors} from '../../libs/Errors.sol';
+import {ILineaAdapter, IBaseAdapter} from './ILineaAdapter.sol';
+import {SafeCast} from 'solidity-utils/contracts/oz-common/SafeCast.sol';
+
+/**
+ * @title LineaAdapter
+ * @author BGD Labs
+ * @notice Linea bridge adapter. Used to send and receive messages cross chain between Ethereum and Linea
+ * @dev it uses the eth balance of CrossChainController contract to pay for message bridging as the method to bridge
+        is called via delegate call
+ * @dev note that this adapter can only be used for the communication path ETHEREUM -> LINEA
+ * @dev documentation regarding the Linea bridge can be found here: https://docs.linea.build/get-started/concepts/message-service#technical-reference
+ */
+contract LineaAdapter is ILineaAdapter, BaseAdapter {
+  /// @inheritdoc ILineaAdapter
+  address public immutable LINEA_MESSAGE_SERVICE;
+
+  /**
+   * @notice only calls from the set message service are accepted.
+   */
+  modifier onlyLineaMessageService() {
+    require(msg.sender == address(LINEA_MESSAGE_SERVICE), Errors.CALLER_NOT_LINEA_MESSAGE_SERVICE);
+    _;
+  }
+
+  /**
+   * @param params object containing the necessary parameters to initialize the contract
+   */
+  constructor(
+    LineaParams memory params
+  )
+    BaseAdapter(
+      params.crossChainController,
+      params.providerGasLimit,
+      'Linea native adapter',
+      params.trustedRemotes
+    )
+  {
+    require(
+      params.lineaMessageService != address(0),
+      Errors.LINEA_MESSAGE_SERVICE_CANT_BE_ADDRESS_0
+    );
+    LINEA_MESSAGE_SERVICE = params.lineaMessageService;
+  }
```

Comment:
don't we need to check here that originChainId is Ethereum chain id? Looks like in another case adapter will be broken forever after creation

Reply:
we check this on deployment proposal, and on receive / forwarding messages.